src/features/time_features.py

```diff
@@ -0,0 +1,55 @@
+import logging
+
+import pandas as pd
+from sklearn.base import BaseEstimator, TransformerMixin
+
+logger = logging.getLogger(__name__)
+
+
+def day_part(hour):
+    if hour in [4, 5]:
+        return "dawn"
+    elif hour in [6, 7]:
+        return "early morning"
+    elif hour in [8, 9, 10]:
+        return "late morning"
+    elif hour in [11, 12, 13]:
+        return "noon"
+    elif hour in [14, 15, 16]:
+        return "afternoon"
+    elif hour in [17, 18, 19]:
+        return "evening"
+    elif hour in [20, 21, 22]:
+        return "night"
+    elif hour in [23, 24, 1, 2, 3]:
+        return "midnight"
+
+
+class GeoDataTransformer(BaseEstimator, TransformerMixin):
+    """
+    Author: Dawid Kowalczyk
+    Date: 31 October 2023
+    Description: Date & Time feature engineering
+    """
+
+    def __init__(self):
+        pass
+
+    def fit(self, X: pd.DataFrame, y: pd.Series = None):
+        return self
+
+    def transform(self, X: pd.DataFrame) -> pd.DataFrame:
+
+        X["trip_duration_seconds"] = (X["dropoff_datetime"] - X["pickup_datetime"]).dt.total_seconds()
```

Comment:
I thought that we discussed that we do not know the dropoff time

Reply:
that's clever, in the case, any variables based on dropoff cannot be crerated

Reply:
(except dropoff lat/lon - which we know), but dropoff datetime - we discussed we don't know